integrations/lib/multiwoven/integrations/destination/zendesk/client.rb

```diff
@@ -0,0 +1,124 @@
+module Multiwoven
+    module Integrations
+        module Destination
+            module Zendesk
+                include Multiwoven::Integrations::Core
+
+                API_VERSION = "59.0"
+
+                class Client < DestinationConnector
+                    prepend Multiwoven::Integrations::Core::RateLimiter
+                    def check_connection(connection_config)
+                        connection_config = connection_config.with_indifferent_access
+                        initialize_client(connection_config)
+                        authenticate_client
+                        success_status
+                      rescue StandardError => e
+                        handle_exception("ZENDESK:CHECK_CONNECTION:EXCEPTION", "error", e)
+                        failure_status(e)
+                    end
+
+                    def discover(_connection_config = nil)
+                        catalog = build_catalog(load_catalog)
+                        catalog.to_multiwoven_message
+                        rescue StandardError => e
+                            handle_exception("ZENDESK:DISCOVER:EXCEPTION", "error", e)
+                            failure_status(e)
+                    end
+
+                    def write(sync_config, tickets, action = "create")
+                        @action = sync_config.stream.action || action
+                        initialize_client(sync_config.destination.connection_specification)
+                        process_tickets(tickets, sync_config.stream)
+                        rescue StandardError => e
+                            handle_exception("ZENDESK:WRITE:EXCEPTION", "error", e)
+                            failure_status(e)
+                    end
+
+                    private
+
+                    def initialize_client(connection_config)
+                        connection_config = connection_config.with_indifferent_access
+                        @client = ZendeskAPI::Client.new do |config|
+                            config.url = ZENDESK_TICKETING_URL
+                            config.username = connection_config[:username]
+                            config.password = connection_config[:password]
+                        end
+                    end
+
+                    def authenticate_client
```

Comment:
Is there an alternative method to check the connection status besides fetching tickets?

Reply:
There are but this seemed like the most straightforward, simplest method.

I could be wrong though. Would you be willing to take a look at the API reference and confirm that I am not missing anything?